index.js

```diff
@@ -11,63 +12,86 @@ const QUESTIONS = [{
         input: "[1,2,3,4,5]",
         output: "5"
     }]
-}];
+}]
 
 
 const SUBMISSION = [
 
 ]
 
-app.post('/signup', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
 
+app.post('/signup', (req, res)=>{
+    // Add logic to decode body
+    // body should have email and password
+    //check if user already exists
+    //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+    // return back 200 status code to the client
 
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+    if(USERS.find(user=>user.username == req.body.username)){
 
+        return res.status(400).json({ error : 'User already exists'})
 
-  // return back 200 status code to the client
-  res.send('Hello World!')
-})
-
-app.post('/login', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+    }else{
 
+    const user = {username: req.body.username, password: req.body.password}
```

Comment:
This can be moved to top of the function and you can check for truthfulness of the username or password fields from API req body